### ISSUE TITLE
no-var-before-return false positive on async code

### ISSUE THREAD

**Issue by artiommorozov** [ajafff/tslint-consistent-codestyle#95]
Following this guideline in case of async functions breaks exception handling.

```
async function x() {
	throw 42;
}

async function y() {
	try {

// bad from lint pov yet correctly prints 42 and 11
		const t = await x();
		return t;

// instant call like this triggers unhandled promise exception as wait occurs 
// outside y() try/catch
//
//		return x();
	} catch (e) {
		console.log(`err: ${e}`);
	}

	return 11;
}

y().then((a) => console.log(a));
```

**Comment by artiommorozov** [comment]
sorry, false alarm
